walk-in-interview/src/main/java/com/google/job/servlets/JobServlet.java

```diff
@@ -128,4 +140,42 @@ private void updateJobPost(String jobId, Job job)
             throw new ServletException(e);
         }
     }
+
+    /**
+     * Returns the Job object.
+     *
+     * @param jobId The job id that corresponds to the job we want to get.
+     * @return Job object with all the details of the job.
+     */
+    private Job fetchJobDetails(String jobId) throws ServletException, ExecutionException, TimeoutException {
+        try {
+            Optional<Job> job = this.jobsDatabase.fetchJob(jobId)
+                    .get(TIMEOUT_SECONDS, TimeUnit.SECONDS);
+
+            if (!job.isPresent()) {
+                throw new IllegalArgumentException("could not find job for this jobId: " + jobId);
+            }
```

Comment:
And you can also handle in the servlet so that you return error 404 resource not found. When a client receive 404, they should know that they're requesting a resource that's not there.